axelor-production/src/main/java/com/axelor/apps/production/service/StockMoveServiceProductionImpl.java

```diff
@@ -142,4 +149,23 @@ public void cancelFromManufOrder(StockMove stockMove, CancelReason cancelReason)
   protected void cancelStockMoveInProduction(StockMove stockMove) throws AxelorException {
     super.cancel(stockMove);
   }
+
+  @Transactional
+  @Override
+  public String realize(StockMove stockMove) throws AxelorException {
+    String newStockSeq = super.realize(stockMove);
+    stockMove
+        .getStockMoveLineList()
+        .forEach(
+            sml -> {
+              if (sml.getProducedManufOrder() != null
+                  && sml.getProducedManufOrder().getSaleOrderLine() != null) {
+                SaleOrderLine saleOrderLine = sml.getProducedManufOrder().getSaleOrderLine();
+                if (saleOrderLine.getProduct().equals(sml.getProduct()))
+                  saleOrderLine.setQtyProduced(sml.getQty());
+                saleOrderLineRepository.save(saleOrderLine);
+              }
+            });
+    return newStockSeq;
+  }
```

Comment:
You should not override this method but instead StockMoveServiceImpl#realizeStockMove which contains only the process linked to the realization. We had issues with transactions and mail services that is why we had to split the method in two.